File: app/src/main/java/kztproject/jp/splacounter/di/FragmentBuilderModule.java
package kztproject.jp.splacounter.di;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import kztproject.jp.splacounter.view.fragment.AuthFragment;
import kztproject.jp.splacounter.view.fragment.RewardDetailFragment;
import kztproject.jp.splacounter.view.fragment.RewardFragment;

@Module
abstract class FragmentBuilderModule {
    @ContributesAndroidInjector
    abstract AuthFragment contributeAuthFragment();

    @ContributesAndroidInjector
    abstract RewardFragment contributeRewardFragment();

    @ContributesAndroidInjector
    abstract RewardDetailFragment contributeRewardAddFragment();
}

File: app/src/main/java/kztproject/jp/splacounter/di/FragmentBuilderModule.kt
package kztproject.jp.splacounter.di

import dagger.Module
import dagger.android.ContributesAndroidInjector
import kztproject.jp.splacounter.reward.detail.ui.RewardDetailFragment
import kztproject.jp.splacounter.reward.list.ui.RewardFragment

@Module
internal abstract class FragmentBuilderModule {

    @ContributesAndroidInjector
    internal abstract fun contributeRewardFragment(): RewardFragment

    @ContributesAndroidInjector
    internal abstract fun contributeRewardAddFragment(): RewardDetailFragment
}
